packages/govuk-frontend/src/govuk/components/header/header.mjs

```diff
@@ -52,22 +52,36 @@ export class Header extends GOVUKFrontendComponent {
     }
 
     this.$module = $module
-    this.$menuButton = $module.querySelector('.govuk-js-header-toggle')
-    this.$menu =
-      this.$menuButton &&
-      $module.querySelector(
-        `#${this.$menuButton.getAttribute('aria-controls')}`
-      )
-
-    if (
-      !(
-        this.$menuButton instanceof HTMLElement ||
-        this.$menu instanceof HTMLElement
-      )
-    ) {
+    const $menuButton = $module.querySelector('.govuk-js-header-toggle')
+
+    // Headers don't necessarily have a navigation.
+    // When they don't, the menu toggle won't be rendered
+    // by our macro (or may be omitted when writing plain HTML)
+    if (!$menuButton) {
       return this
     }
 
+    const $menu = document.getElementById(
+      $menuButton.getAttribute('aria-controls')
+    )
```

Comment:
Would you mind if we added the guard in https://github.com/alphagov/govuk-frontend/pull/4206#discussion_r1334120152?

Matches the approach we recently added to **Skip link**

```suggestion
    const menuId = $menuButton.getAttribute('aria-controls')

    // Check for menu button target
    if (!menuId) {
      throw new ElementError(this.$module, {
        componentName: 'Header',
        identifier: '$menuButton["aria-controls"]',
        expectedType: 'string'
      })
    }

    const $menu = document.getElementById(menuId)
```